f8a_worker/workers/new_graph_importer.py

```diff
@@ -16,7 +16,43 @@
     endpoint=_SELECTIVE_SERVICE_ENDPOINT)
 
 
-class NewGraphImporterTask(BaseTask):
+class NewPackageGraphImporterTask(BaseTask):
+    """Ingest to graph node."""
+
+    def execute(self, arguments):
+        """Task code.
+
+        :param arguments: dictionary with task arguments
+        :return: {}, results
+        """
+        self._strict_assert(arguments.get('ecosystem'))
+        self._strict_assert(arguments.get('name'))
+
+        package_list = [{
+                'ecosystem': arguments.get('ecosystem'),
+                'name': arguments['name'],
+                'version': arguments.get('version'),
```

Comment:
There are two flows, 
1) **newPackageFlow** -> **NewInitPackageFlow** -> **NewMetaDataTask** -> **newPackageAnalysisFlow** -> **NewPackageGraphImporterTask**
2) **newPackageAnalysisFlow** -> **NewInitPackageAnlysisFlow** -> **NewGithubDetails** -> **NewPackageAnalysisGraphImporterTask**

As you can see here and in flow definitions as well, each flow has one graph importer task, 
**NewPackageAnalysisGraphImporterTask** 
**NewPackageGraphImporterTask**

If it a full ingestion both the flows will be executed, in this case the parent flow "newPackageFlow" will require version to create version node.
In case of Github refresh, only the child flow "newPackageAnalysisFlow "will be called , which does not need version.
The task you are pointing is NewPackageGraphImporterTask and it belongs to parent flow which requires version, this task wont be used in github refresh.

Reply:
The problem i see here is, during the newPackageFlow, it will call graphimportertask twice. Once for pkg analysis flow and once for package flow. Is that correct? why do we need it twice?

Reply:
It is because of Data importer, it requires name of file to be ingested into graph. In S3 we have some files having package level data and some having version level data. In case of full ingestion both level's data has to be ingested and for github refresh, data of github file has to be ingested, so we have two tasks one which calls data importer API for package level data ingestion and other which calls API for version level data ingestion.